smart-contracts/contracts/PublicLock.sol

```diff
@@ -221,20 +221,42 @@ contract PublicLock is ILockCore, ERC165, IERC721, IERC721Receiver, Ownable {
   }
 
   /**
-   * @dev Called by owner to wiwthdraw all funds from the lock.
+   * @dev Called by owner to withdraw all funds from the lock.
    * TODO: consider allowing anybody to trigger this as long as it goes to owner anyway?
-   * TODO: consider partial withdraws?
    * TODO: check for re-entrency?
    */
-  function withdraw(
-  )
+  function withdraw()
     external
     onlyOwner
   {
     uint256 balance = address(this).balance;
     require(balance > 0, "Not enough funds");
+    _withdraw(balance);
+  }
+
+  /**
+   * @dev Called by owner to partially withdraw funds from the lock.
+   * TODO: consider allowing anybody to trigger this as long as it goes to owner anyway?
+   * TODO: check for re-entrency?
+   */
+  function partialWithdraw(uint _amount)
+    external
+    onlyOwner
+  {
+    uint256 balance = address(this).balance;
+    require(balance > 0 && balance > _amount, "Not enough funds");
```

Comment:
balance >= _amount (not just greater than)... unless you prefer forcing them to use the other method when withdrawing all.

Reply:
well, semantically it's a bit weird to use the `partialWithdraw` function to withdraw all the funds. I'd rather let each withdraw function do one thing, but am open to modifying this if you or @julien51 feel strongly otherwise...

Reply:
I agree that it is weird but it may be confusing to people to return an error when in practice I think it does not change much for us? I'd be liberal in what we expect from people and conservative in what we do ;)

Reply:
sounds good, I'll modify now.

Reply:
done.